test/test_pubsub.py

```diff
@@ -99,3 +106,41 @@ async def test_unsubscribe_sub_id_not_exists(mock_pubsub_subscriptions):
     assert len(mock_pubsub_subscriptions._subscriptions) == 1
     assert 1 in mock_pubsub_subscriptions._subscriptions
     assert result is False
+
+
+@pytest.fixture()
+def mock_pubsub_publish(mocker):
+    pubsub = PubSub()
+    redis_mock = fakeredis.aioredis.FakeRedis()
+    mocker.patch.object(pubsub, '_redis', redis_mock)
+    mocker.patch.object(pubsub._redis, 'execute_command')
+    return pubsub
+
+
+@pytest.mark.asyncio
+async def test_pubsub_publish_couldevent(mock_pubsub_publish):
+    """
+    Test Case: Validate and check the json built by the cloud event
+    published in the channel by the redis publisher.
+
+    Expected Results:
+        Validate that a json is sent to the channel and assert the json values from
+        data and attributes parameters in Pubsub.publish_cloudevent(). There's no
+        return value, but a json to be published in a channel.
+    """
+
+    data = 'validate json'
+    attributes = { "specversion": "1.0",  "id": "6878b661-96dc-4e93-8c92-26eb9ff8db64", "source": "https://api.kernelci.org/", "type": "api.kernelci.org", "time": "2022-01-31T21:29:29.675593+00:00"}
+
+    await mock_pubsub_publish.publish_cloudevent('CHANNEL1', data, attributes)
+
+    expected_json = str.encode('{"specversion": "1.0", "id": "6878b661-96dc-4e93-8c92-26eb9ff8db64", "source": "https://api.kernelci.org/", "type": "api.kernelci.org", "time": "2022-01-31T21:29:29.675593+00:00", "data": "validate json"}')
+
+    json_arg = mock_pubsub_publish._redis.execute_command.call_args.args[2]
+
+    try:
+        json.loads(json_arg)
+    except json.JSONDecodeError:
+        assert False
```

Comment:
I don't quite get the idea of `assert False` here.
If `json_arg` is supposed to be a valid JSON object there should be neither `try..except` block nor `assert False`. 
If for some reason `json.loads` raises an exception the test should end up with `Error` status while `assert False` will mark it as `Failed`.

Reply:
thanks for noticing this, you're right, so I removed the `try..except` block and now the test will just fail in case of any problem with the JSON.